### PR TITLE
Allow INSERT null for SQL Server varbinary type

### LINKED ISSUE #4795: Failed to insert NULL for varbinary in SQL Server
Steps to reproduce:
```sql
# SQL Server
1> create table dbo.test (c1 int, c2 varbinary(max), c3 image);
2> go

# Presto
presto> insert into sqlserver.dbo.test values (1, X'65683F', X'65683F');
INSERT: 1 row

presto> insert into sqlserver.dbo.test values (1, null, null);

Query 20200812_140614_00007_yrg9g failed: Failed to insert data: Implicit conversion from data type nvarchar to varbinary(max) is not allowed. Use the CONVERT function to run this query.
io.prestosql.spi.PrestoException: Failed to insert data: Implicit conversion from data type nvarchar to varbinary(max) is not allowed. Use the CONVERT function to run this query.
	at io.prestosql.plugin.jdbc.JdbcPageSink.finish(JdbcPageSink.java:168)
	at io.prestosql.operator.TableWriterOperator.finish(TableWriterOperator.java:203)
	at io.prestosql.operator.Driver.processInternal(Driver.java:397)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1076)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_testversion____20200812_135122_1.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.sql.BatchUpdateException: Implicit conversion from data type nvarchar to varbinary(max) is not allowed. Use the CONVERT function to run this query.
	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.executeBatch(SQLServerPreparedStatement.java:2065)
	at io.prestosql.plugin.jdbc.JdbcPageSink.finish(JdbcPageSink.java:160)
	... 12 more
```

### PR DESCRIPTION
**Pull request by ebyhr**
Fixes #4795

The original `statement.setObject(index, null)` in WriteFunction causes conversion error.

### CODE REVIEW

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java [hunk 1]
```diff
@@ -111,6 +111,18 @@ public void testDecimalPredicatePushdown()
...
     }
 
+    @Test
+    public void testInsertVarbinary()
+            throws Exception
+    {
+        try (AutoCloseable ignoreTable = withTable("test_varbinary",
+                "(x varbinary(max))")) {
+            assertUpdate("INSERT INTO test_varbinary VALUES (X'65683F'), (NULL)", 2);
+            assertQuery("SELECT * FROM test_varbinary",
```
Comment (findepi) [inline presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java:121]: Can this be tested with `DataTypeTest`? Maybe we can have `TestSqlServerTypeMapping` stub?
(see https://github.com/prestosql/presto/issues/565)

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java [hunk 2]
```diff
@@ -111,6 +111,18 @@ public void testDecimalPredicatePushdown()
...
+                    "VALUES (X'65683F'), (NULL)");
```
Comment (findepi) [inline presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java:122]: no need for a newline

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java [hunk 3]
```diff
@@ -111,6 +111,18 @@ public void testDecimalPredicatePushdown()
         }
     }
 
+    @Test
+    public void testInsertVarbinary()
+            throws Exception
+    {
+        try (AutoCloseable ignoreTable = withTable("test_varbinary",
+                "(x varbinary(max))")) {
```
Comment (findepi) [inline presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java:119]: nit: no need for a newline

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java [hunk 4]
```diff
@@ -111,6 +111,18 @@ public void testDecimalPredicatePushdown()
         }
     }
 
+    @Test
+    public void testInsertVarbinary()
```
Comment (findepi) [inline presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java:115]: This test seems to be passing even without changes in SqlServerClient. 
Intentional?

Comment (ebyhr) [reply presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java]: @findepi Sorry, it's unintentional. I should have separated INSERT statements.

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java (hunk 1 above)
Comment (ebyhr) [reply presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java]: @findepi I avoided `TestSqlServerTypeMapping` considering #4734, but moved the test to the class.

Comment (findepi) [reply presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java]: Right. Let's have a TODO in the code, to move the method over there, while `TestSqlServerTypeMapping` exists

Comment (findepi) [reply presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java]: Oh, i see you called the test class `TestSqlServerTypeMappi`. Fine.

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerTypeMapping.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                .execute(getQueryRunner(), prestoCreateAsSelect("test_varbinary"));
+    }
+
+    private static DataTypeTest varbinaryTestCases(DataType<byte[]> varbinaryDataType)
+    {
+        return DataTypeTest.create()
+                .addRoundTrip(varbinaryDataType, "hello".getBytes(UTF_8))
+                .addRoundTrip(varbinaryDataType, "Piękna łąka w 東京都".getBytes(UTF_8))
+                .addRoundTrip(varbinaryDataType, "Bag full of 💰".getBytes(UTF_16LE))
+                .addRoundTrip(varbinaryDataType, null)
```
Comment (findepi) [inline presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerTypeMapping.java:70]: make this first

File: presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerTypeMapping.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Test
+    public void testVarbinary()
+    {
+        varbinaryTestCases(varbinaryDataType())
+                .execute(getQueryRunner(), prestoCreateAsSelect("test_varbinary"));
+    }
+
+    private static DataTypeTest varbinaryTestCases(DataType<byte[]> varbinaryDataType)
```
Comment (findepi) [inline presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerTypeMapping.java:64]: no need for a method, you can inline with test body

**Review by findepi — APPROVED** [review]

**Pull request merged**